pkg/cmd/roachtest/cluster.go

```diff
@@ -209,7 +209,17 @@ func unregisterCluster(c *cluster) bool {
 	return exists
 }
 
-func execCmd(ctx context.Context, l *logger, args ...string) error {
+// execCmd runs a command. stdout/stderr can be passed in to capture the
+// commands's output. If nil, the logger's stdout/stderr will be used.
+func execCmd(ctx context.Context, l *logger, stdout, stderr io.Writer, args ...string) error {
+	if (stdout == nil) != (stderr == nil) {
+		log.Fatal(ctx, "can't specify only one of stdout/stderr")
+	}
+	if stdout == nil {
+		stdout = l.stdout
+		stderr = l.stderr
```

Comment:
Should this be in a separate `if stderr == nil` block? Sort of weird that `stderr` gets overwritten unconditionally if `stdout == nil`.